metastore/scripts/upgrade/derby/057-HIVE-18193.derby.sql

```diff
@@ -0,0 +1,24 @@
+
+-- Populate NEXT_WRITE_ID for each Transactional table and set next write ID same as next txn ID
+INSERT INTO NEXT_WRITE_ID (NWI_DATABASE, NWI_TABLE, NWI_NEXT)
+    SELECT * FROM
+        (SELECT DB.NAME, TBL_INFO.TBL_NAME FROM DBS DB,
+            (SELECT TBL.DB_ID, TBL.TBL_NAME FROM TBLS TBL,
+                (SELECT TBL_ID FROM TABLE_PARAMS WHERE PARAM_KEY='transactional' AND CAST(PARAM_VALUE AS VARCHAR(128))='true') TBL_PARAM
+            WHERE TBL.TBL_ID=TBL_PARAM.TBL_ID) TBL_INFO
+        where DB.DB_ID=TBL_INFO.DB_ID) DB_TBL_NAME,
+        (SELECT NTXN_NEXT FROM NEXT_TXN_ID) NEXT_TXN_ID;
+
+-- Populate TXN_TO_WRITE_ID for each aborted/open txns and set write ID equal to txn ID
+INSERT INTO TXN_TO_WRITE_ID (T2W_DATABASE, T2W_TABLE, T2W_TXNID, T2W_WRITEID)
+    SELECT * FROM
+        (SELECT DB.NAME, TBL_INFO.TBL_NAME FROM DBS DB,
```

Comment:
a temp table / materialized view can be created with the db_name, table_name for all acid tables and that can be used for populating both NEXT_WRITE_ID and TXN_TO_WRITE_ID tables.

Reply:
Number of tables is not so huge and hence no need to cache the joined results.